emitterc.go

```diff
@@ -1980,9 +1980,17 @@ func yaml_emitter_write_folded_scalar(emitter *yaml_emitter_t, value []byte) boo
 	return true
 }
 
-func yaml_emitter_write_comment(emitter *yaml_emitter_t, comment []byte) bool {
+func yaml_emitter_write_comment(emitter *yaml_emitter_t, comment []byte, padding int) bool {
 	breaks := false
 	pound := false
+
+	// [Go] Start by adding any additional padding to the line comment.
+	for i := 0; i < padding; i++ {
+		if !put(emitter, ' ') {
+			return false
+		}
+	}
+
```

Comment:
Any reason to inject the extra spaces here instead of directly in `yaml_emitter_process_line_comment` ?
That would avoid having to provide a `0` any time `yaml_emitter_write_comment` is called

https://github.com/braydonk/yaml/blob/875dd3d1bbde541bdaf2879765b636fea71e0e42/emitterc.go#L1149-L1153

Reply:
The main reason is that then I would have to change the logic at line 2007 which checks for the beginning of the line comment to be a `#`, and if it's not then it inserts `# `. It's sorta weird logic imo, but in general I tend to prefer stuff like this that simply adds an override to existing logic over messing with it directly. Just want to keep my involvement with the original library to a minimum.

Reply:
Got it ! My understanding of this logic is that it applies to non-line comments, since line comments always have `comment[0] == "#"` and never have breaks, so I wouldn't have to worry much about this logic for anything related to inserting spaces before writing the comment itself

I have been experimenting with this alternative implementation https://github.com/badouralix/go-yaml/commit/f2c7845329cefd433ad406d8a340eb8e238fdc08 and couldn't find any edge case breaking the behavior of line comments
One interesting thing is that the config is explicitly `2` to insert 2 spaces before a line comment, whereas the padding config wants `1` to insert 2 spaces
Wouldn't it be easier to keep an alignment between the config and the number of spaces rather than having a discrepancy ?

Reply:
The reason I had kept with the idea of "padding" and making it a padding of 1 space over the 1 space that automatically gets inserted. But I can see how that would be harder to explain that in documentation vs the default of 1.

That being said, I like your implementation better. I have a couple of comments, but I think overall yours is better and I'd rather accept that in favour of my version. Could you open a PR with it and add me as a reviewer?

Reply:
Just opened it https://github.com/braydonk/yaml/pull/16 but it seems I cannot add you as a reviewer 🙈

Reply:
Thanks! Not sure why it doesn't let you add me as a reviewer, but I just requested myself instead.